### PR TITLE
Add OpenAPI to TypeScript and OpenAPI schema generation CLI commands

### PR DESCRIPTION
**Pull request by Goldziher**
# PR Checklist

- [x] Have you followed the guidelines in `CONTRIBUTING.md`?
- [x] Have you got 100% test coverage on new code?
- [x] Have you updated the prose documentation?
- [x] Have you updated the reference documentation?

### CODE REVIEW

**Comment by provinzkraut** [comment]
Should this not go into `contrib`?

**Comment by Goldziher** [comment]
> Should this not go into `contrib`?

There are no external dependencies here, it's going to go into the CLI

File: pyproject.toml [hunk 1]
```diff
@@ -58,9 +58,10 @@ typing-extensions = "*"
 msgspec = ">=0.11.0"
 jinja2 = { version = ">=3.1.2", optional = true }
 mako = { version = ">=1.2.4", optional = true }
-click = {version = ">=8.1.3", optional = true}
-rich = {version = ">=13.0.1", optional = true}
-importlib-metadata = { version = ">=5.2.0", python = "<3.10" }
+click = {version = "*", optional = true}
+rich = {version = "*", optional = true}
```
Comment (provinzkraut) [inline pyproject.toml:62]:
```suggestion
rich = {version = ">=13.0.0", optional = true}
```

This is required. Can't pinpoint what it was, but something with the tables caused issues in earlier versions

File: starlite/cli/commands/openapi.py [hunk 2]
```diff
@@ -0,0 +1,53 @@
...
+    if not app.openapi_schema:
+        raise StarliteCLIException("Starlite application does not have an OpenAPI schema")
+
+    if output.lower().endswith(("yml", "yaml")):
+        content = dump_yaml(app.openapi_schema.dict(by_alias=True, exclude_none=True), default_flow_style=False)
+    else:
+        content = dumps(app.openapi_schema.dict(by_alias=True, exclude_none=True), indent=4)
+
+    try:
+        Path(output).write_bytes(content.encode())
```
Comment (provinzkraut) [inline starlite/cli/commands/openapi.py:35]:
```suggestion
        Path(output).write_text(content)
```
?

File: starlite/cli/commands/openapi.py [hunk 3]
```diff
@@ -0,0 +1,53 @@
...
+beautifier = Beautifier()
+
+
+@group(cls=StarliteGroup, name="openapi")
+def openapi_group() -> None:
+    """Manage server-side openapi."""
+
+
+@openapi_group.command("schema")
+@option("--output", help="output file path", type=str, default="openapi_schema.json")
```
Comment (provinzkraut) [inline starlite/cli/commands/openapi.py:23]:
```suggestion
@option("--output", help="output file path", type=click.Path(), default="openapi_schema.json", show_default=True)
```
Ensures that it's a valid path and shows default

File: starlite/cli/commands/openapi.py [hunk 4]
```diff
@@ -0,0 +1,53 @@
...
+        content = dumps(app.openapi_schema.dict(by_alias=True, exclude_none=True), indent=4)
+
+    try:
+        Path(output).write_bytes(content.encode())
+    except OSError as e:
+        raise StarliteCLIException(f"failed to write schema to path {output}") from e
+
+
+@openapi_group.command("typescript")
+@option("--output", help="output file path", type=str, default="api-specs.ts")
```
Comment (provinzkraut) [inline starlite/cli/commands/openapi.py:41]:
```suggestion
@option("--output", help="output file path", type=click.Path(), default="api-specs.ts", show_default=True)
```

File: starlite/cli/commands/openapi.py [hunk 5]
```diff
@@ -0,0 +1,53 @@
...
+@option("--namespace", help="namespace to use for the typescript specs", type=str, default="API")
+def generate_typescript_specs(app: Starlite, output: str, namespace: str) -> None:
+    """Generate TypeScript specs from the OpenAPI schema."""
+    if not app.openapi_schema:
+        raise StarliteCLIException("Starlite application does not have an OpenAPI schema")
+
+    try:
+        specs = convert_openapi_to_typescript(app.openapi_schema, namespace)
+        beautified_output = beautifier.beautify(specs.write())
+        Path(output).write_bytes(beautified_output.encode())
```
Comment (provinzkraut) [inline starlite/cli/commands/openapi.py:51]:
```suggestion
        Path(output).write_text(beautified_output)
```

**Review by provinzkraut — COMMENTED** [review]
Looks really nice! Just a few minor comments on the CLI stuff

**Review by provinzkraut — APPROVED** [review]

**Comment by provinzkraut** [comment]
Can you fix the missing coverage? Looks to be just one line

**Pull request merged**